test/wh_test_she.c

```diff
@@ -0,0 +1,385 @@
+        goto exit;
+    }
+    /* generate a fake bootloader */
+    if ((ret = wc_RNG_GenerateBlock(rng, bootloader, sizeof(bootloader))) != 0) {
+        WH_ERROR_PRINT("Failed to wc_RNG_GenerateBlock %d\n", ret);
+        goto exit;
+    }
+    /* cmac 0..0 | size | bootloader */
+    if ((ret = wc_InitCmac(cmac, key, sizeof(key), WC_CMAC_AES, NULL)) != 0) {
+        WH_ERROR_PRINT("Failed to wc_InitCmac %d\n", ret);
+        goto exit;
+    }
+    if ((ret = wc_CmacUpdate(cmac, zeros, sizeof(zeros))) != 0) {
+        WH_ERROR_PRINT("Failed to wc_CmacUpdate %d\n", ret);
+        goto exit;
+    }
+    if ((ret = wc_CmacUpdate(cmac, (uint8_t*)&bootloaderSz, sizeof(bootloaderSz))) != 0) {
+        WH_ERROR_PRINT("Failed to wc_CmacUpdate %d\n", ret);
+        goto exit;
+    }
+    if ((ret = wc_CmacUpdate(cmac, bootloader, sizeof(bootloader))) != 0) {
+        WH_ERROR_PRINT("Failed to wc_CmacUpdate %d\n", ret);
+        goto exit;
+    }
+    digestSz = AES_BLOCK_SIZE;
+    if ((ret = wc_CmacFinal(cmac, keyAndDigest + sizeof(key), &digestSz)) != 0) {
+        WH_ERROR_PRINT("Failed to wc_CmacFinal %d\n", ret);
+        goto exit;
+    }
+    memcpy(keyAndDigest, key, sizeof(key));
+    /* store cmac key and digest */
+    if ((ret = wh_Client_NvmAddObject(client, MAKE_WOLFHSM_KEYID(WOLFHSM_KEYTYPE_SHE, client->comm->client_id, WOLFHSM_SHE_BOOT_MAC_KEY_ID), 0, 0, 0, NULL, sizeof(keyAndDigest), keyAndDigest, &outRc)) != 0 || outRc != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_NvmAddObjectRequest %d\n", ret);
+        goto exit;
+    }
+    /* store the vector master ecu key */
+    if ((ret = wh_Client_NvmAddObject(client, MAKE_WOLFHSM_KEYID(WOLFHSM_KEYTYPE_SHE, client->comm->client_id, WOLFHSM_SHE_MASTER_ECU_KEY_ID), 0, 0, 0, NULL, sizeof(vectorMasterEcuKey), vectorMasterEcuKey, &outRc)) != 0 || outRc != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_NvmAddObjectRequest %d\n", ret);
+        goto exit;
+    }
+    /* store the secret key */
+    if ((ret = wh_Client_NvmAddObject(client, MAKE_WOLFHSM_KEYID(WOLFHSM_KEYTYPE_SHE, client->comm->client_id, WOLFHSM_SHE_SECRET_KEY_ID), 0, 0, 0, NULL, sizeof(secretKey), secretKey, &outRc)) != 0 || outRc != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_NvmAddObjectRequest %d\n", ret);
+        goto exit;
+    }
+    /* store the prng seed */
+    if ((ret = wh_Client_NvmAddObject(client, MAKE_WOLFHSM_KEYID(WOLFHSM_KEYTYPE_SHE, client->comm->client_id, WOLFHSM_SHE_PRNG_SEED_ID), 0, 0, 0, NULL, sizeof(prngSeed), prngSeed, &outRc)) != 0 || outRc != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_NvmAddObjectRequest %d\n", ret);
+        goto exit;
+    }
+    /* set the she uid */
+    if ((ret = wh_Client_SheSetUid(client, sheUid, sizeof(sheUid))) != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_SheSetUid %d\n", ret);
+        goto exit;
+    }
+    /* verify bootloader */
+    if ((ret = wh_Client_SheSecureBoot(client, bootloader, bootloaderSz)) != 0) {
```

Comment:
can we add a negative test to know that the validation works and wont validate everything?